ouroboros-consensus/test-consensus/Test/Consensus/Mempool.hs

```diff
@@ -124,6 +128,128 @@ prop_Mempool_InvalidTxsNeverAdded setup =
         , txInMempool `notElem` txsInMempoolBefore
         ]
 
+-- | When the mempool is at capacity, test that 'addTxs' blocks when
+-- attempting to add one more transaction and that it unblocks when there is
+-- adequate space. Adequate space is made by adding all of the existing
+-- transactions to the ledger and removing them from the mempool.
+--
+-- Ignore the "100% empty Mempool" label in the test output, that is there
+-- because we reuse 'withTestMempool' and always start with an empty Mempool
+-- and 'LedgerState'.
+prop_Mempool_Capacity :: MempoolCapTestSetup -> Property
```

Comment:
(You have probably mentioned this already in one of your comments:) This only blocks when adding the *l*ast transaction. We also want to test what happens if multiple transactions cannot yet be added to the Mempool, that we need to block, and that they are added one by one. It is there that a lot of the complexity of `implAddTxs` is located.